pkg/shell/cockpit-docker.js

```diff
@@ -684,40 +701,126 @@ PageRunImage.prototype = {
         $("#containers-run-image-name").val(make_name());
         $("#containers-run-image-command").val(quote_cmdline(PageRunImage.image_info.config.Cmd));
 
-        function render_port(p) {
-            var port_input = $('<input class="form-control" style="display:inline;width:auto" >');
-            var tr =
-                $('<tr class="port-map">').append(
-                    $('<td>').text(
-                        cockpit.format(_("Bind port $0 to "), p)),
-                    $('<td colspan="2">').append(port_input));
+        /* render a port, all parameters are optional and default is to let internal port be editable */
+        function render_port(port_internal, port_protocol, port_internal_editable, port_external) {
+            if (port_internal === undefined)
+                port_internal = '';
+            if (port_protocol === undefined)
+                port_protocol = 'TCP';
+            if (port_internal_editable === undefined)
+                port_internal_editable = true;
+            if (port_external === undefined)
+                port_external = '';
+
+            var port_internal_element = $('<input type="text" class="form-control">');
+            var port_external_element = $('<input type="text" class="form-control">');
+
+            var protocol_select = $('<select class="form-control selectpicker" data-width="60px">')
+                        .append($('<option>').text(_("TCP")))
+                        .append($('<option>').text(_("UDP")));
+
+            var mapping_add = $('<button type="button" class="btn btn-default fa fa-plus">');
+            var mapping_remove = $('<button type="button" class="btn btn-default pficon-close">');
+
+            /* if internal port is editable, allow row delete, otherwise disable modification */
+            if (port_internal_editable) {
+                mapping_remove.on('click', function () {
+                    var parent = $(this).closest("form");
+                    /* don't remove last row */
+                    if (parent.parent().children().length > 1)
+                        parent.remove();
+                    else {
```

Comment:
If one condition has braces, I think we should make them all have braces. It's far more readable that way. WDYT?